### PR TITLE
Convert c-type cast to C++ cast

### PR DESCRIPTION
**Pull request by ggfan**
this should get rid of issue https://github.com/google/oboe/issues/169. PTAL

### CODE REVIEW

File: src/common/Utilities.cpp [hunk 1]
```diff
@@ -36,12 +36,12 @@ void convertFloatToPcm16(const float *source, int16_t *destination, int32_t numS
         float fval = source[i];
         fval += 1.0; // to avoid discontinuity at 0.0 caused by truncation
         fval *= 32768.0f;
-        auto sample = (int32_t) fval;
+        auto sample = static_cast<int32_t>(fval + 0.5f);
```
Comment (philburk) [inline src/common/Utilities.cpp]: This changes the behavior of the code from floor to round! Please remove the "+ 0.5f". 
Maybe it should match the similar code in:
"frameworks/av/media/libaaudio/src/utility/AAudioUtilities.cpp"
If so then, let's do that in a separate Pull Request.

File: src/opensles/AudioStreamOpenSLES.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ SLuint32 AudioStreamOpenSLES::channelCountToChannelMaskDefault(int channelCount)
 
 static bool s_isLittleEndian() {
     static uint32_t value = 1;
-    return *((uint8_t *) &value) == 1; // Does address point to LSB?
+    return (*reinterpret_cast<uint8_t *>(&value) == 1);
```
Comment (philburk) [inline src/opensles/AudioStreamOpenSLES.cpp]: Please restore the comment.

File: src/common/Utilities.cpp (hunk 1 above)
Comment (ggfan) [reply src/common/Utilities.cpp]: yeah, it is: so please check whether this helps or not ( rounding up or cut-off ). Let me restore it now

Comment (ggfan) [reply src/common/Utilities.cpp]: done reverting, thanks

File: src/opensles/AudioStreamOpenSLES.cpp (hunk 2 above)
Comment (ggfan) [reply src/opensles/AudioStreamOpenSLES.cpp]: done, thanks! ( In old days, comment on the "return..." statement caused compile errors, I was under that impression.)  pointer is a special kind of value, once value is LE, pointer should be LE already, correct?

Comment (philburk) [reply src/opensles/AudioStreamOpenSLES.cpp]: The pointer is not little endian. It points to a value that is either big or little endian. This is a trick for querying endianness.

**Comment by philburk** [comment]
Thanks for doing this. But when making multiple changes in one Pull Request, please do **not** use "commit --amend".  Just keep adding commits to the same branch. It makes it very hard to see changes between patch sets.
